documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,234 @@
+        </ul>
+    </li>
+  </ol>
+</details>
+<!--/TABLE OF CONTENTS-->
+
+<!--VISION-->
+## Vision
+
+"Department and Employee Management" is a web application that allows an administrator to record information about a department and an employee.
+
+Application should provide:
+
+* Storing list of department and list of employee in a database;
+* Display list of department and the average salary for these departments;
+* Updating the list of department (adding, editing, removing);
+* Display list of employee in the departments with an indication of the salary for each employee;
+* Updating the list of employee (adding, editing, removing);
+* Filtering by employees born on a certain date or in the period between dates.
+<!--/VISION-->
+
+<!--DEPARTMENT-->
+## Department
+### Display list of department
+
+The mode is designed to view the list of department.
+
+<b>Main scenario:</b>
+* User selects item “Department”;
+* Application displays list of Department.
+
+![](department_listt.jpg)
+Pic. 2.1 View the Department list.
+
+The list displays the following columns:
+* ID – unique department number;
+* Nubmer of employees - the number of employees in this department
+* The average salary ($) - average salary of all employees of the department
+### Add department
+
+<b>Main scenario:</b>
+* Administrator clicks the “Add” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying;
+* If new department record is successfully added, then list of departments with added records is displaying.
+
+<b>Cancel operation scenario:</b>
+* Administrator clicks the “Add” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Cancel” button;
+* Data don’t save in data base, then list of departments records is displaying to administrator.
+* If the administrator selects the menu item "Department ”or "Employee", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](add_department.jpg)
+<br>Pic. 2.2 Add department.
+
+<b>When adding a department, the following details are entered:</b>
+* Name department - name department
+* Add employee - adding department employees(optional field)
+### Edit department
+
+<b>Main scenario:</b>
+* Administrator clicks the “Edit” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator edit department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then edited data is added to database;
+* If error occurs, then error message is displaying;
```

Comment:
It would be great to see a mockup with error messages.

Reply:
Fixed